saleor/plugins/webhook/tasks.py

```diff
@@ -144,27 +150,72 @@ def send_webhook_using_google_cloud_pubsub(
     )
 
 
+def _send_webhook_to_target(parts, target_url, message, domain, signature, event_type):
+    scheme = parts.scheme.lower()
+    if scheme in [WebhookSchemes.HTTP, WebhookSchemes.HTTPS]:
+        send_webhook_using_http(target_url, message, domain, signature, event_type)
+    elif scheme == WebhookSchemes.AWS_SQS:
+        send_webhook_using_aws_sqs(target_url, message, domain, signature, event_type)
+    elif scheme == WebhookSchemes.GOOGLE_CLOUD_PUBSUB:
+        send_webhook_using_google_cloud_pubsub(
+            target_url, message, domain, signature, event_type
+        )
+    else:
+        raise ValueError("Unknown webhook scheme: %r" % (parts.scheme,))
+
+
 @app.task(
+    bind=True,
     autoretry_for=(RequestException,),
     retry_backoff=10,
     retry_kwargs={"max_retries": 5},
     compression="zlib",
 )
-def send_webhook_request(webhook_id, target_url, secret, event_type, data):
+def send_webhook_request(
+    self, webhook_id, target_url, secret, event_type, event_payload_id
+):
+    try:
+        data = EventPayload.objects.get(id=event_payload_id)
+    except EventPayload.DoesNotExist:
+        task_logger.error(
+            "Cannot find payload related to webhook.",
+            extra={
+                "webhook_id": webhook_id,
+                "event_payload_id": event_payload_id,
+                "target_url": target_url,
+                "event_type": event_type,
+            },
+        )
+        return
+
     parts = urlparse(target_url)
     domain = Site.objects.get_current().domain
-    message = data.encode("utf-8")
+    message = data.payload.encode("utf-8")
     signature = signature_for_payload(message, secret)
-    if parts.scheme.lower() in [WebhookSchemes.HTTP, WebhookSchemes.HTTPS]:
-        send_webhook_using_http(target_url, message, domain, signature, event_type)
-    elif parts.scheme.lower() == WebhookSchemes.AWS_SQS:
-        send_webhook_using_aws_sqs(target_url, message, domain, signature, event_type)
-    elif parts.scheme.lower() == WebhookSchemes.GOOGLE_CLOUD_PUBSUB:
-        send_webhook_using_google_cloud_pubsub(
-            target_url, message, domain, signature, event_type
+    with _catch_duration_time() as duration:
+        try:
+            _send_webhook_to_target(
+                parts, target_url, message, domain, signature, event_type
+            )
+        except Exception as exc:
```

Comment:
Maybe we can convert these two calls into one?
I mean, almost all parameters are the same, you can assign an exception to error variable, and assign JobStatus based on this. Then if an error exists, just raise excpetion. What do you think?